src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/IHeartbeatProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Collections.Generic;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    internal interface IHeartbeatProvider : IHeartbeatPropertyManager, IDisposable
+    {
+        string InstrumentationKey { get; set; }
+
+        void Initialize(TelemetryConfiguration configuration);
```

Comment:
you can inherit from `ITelemetryModule` interface to get this method defined. It would be better for consistency